modules/default-flatpaks/v2/post-boot/system-flatpak-setup

```diff
@@ -0,0 +1,41 @@
+#!/usr/bin/env nu
+
+const usrSharePath = "/usr/share/bluebuild/default-flatpaks"
+const configPath = $"($usrSharePath)/configuration.yaml"
+
+def main [] {
+    let configFile = open $configPath
+
+    if (flatpak remotes | str contains fedora) {
```

Comment:
I don't know how to do this in nushell, but to make this more robust, we need to:

List system & user remotes, while not executing `--user` (not allowed with root, so `flatpak remotes --user` is not possible):

```bash
FLATPAK_SYSTEM_REMOTES=($(flatpak remotes | awk '$2 == "system" {print $1}'))
FLATPAK_USER_REMOTES=($(flatpak remotes | awk '$2 == "user" {print $1}'))
```

then determine if `fedora` & `fedora-testing` repo exist in both `--system` & `--user` remotes:

```bash
for system_remote in "${FLATPAK_SYSTEM_REMOTES[@]}"; do
  if [[ $(echo "${system_remote}" | awk '/^fedora$/') == "fedora" ]]; then
    fedora_system=true
  fi  
done
for system_remote in "${FLATPAK_SYSTEM_REMOTES[@]}"; do
  if [[ $(echo "${system_remote}" | awk '/^fedora-testing$/') == "fedora-testing" ]]; then
    fedora_testing_user=true
  fi  
done

for user_remote in "${FLATPAK_USER_REMOTES[@]}"; do
  if [[ $(echo "${user_remote}" | awk '/^fedora$/') == "fedora" ]]; then
    fedora_user=true
  fi  
done
for user_remote in "${FLATPAK_USER_REMOTES[@]}"; do
  if [[ $(echo "${user_remote}" | awk '/^fedora-testing$/') == "fedora-testing" ]]; then
    fedora_testing_user=true
  fi  
done
```

then do

```bash
if "${fedora_system}" || "${fedora_testing_system}" || "${fedora_user}" || "${fedora_testing_user}" ; then
  /usr/bin/gnome-software --quit
  /usr/lib/fedora-third-party/fedora-third-party-opt-out
  /usr/bin/fedora-third-party disable
fi

if "${fedora_system}"; then
  flatpak remote-delete --system fedora --force
fi
if "${fedora_system_testing}"; then  
  flatpak remote-delete --system fedora-testing --force
fi
```

Reply:
Isn't it possible to just check `flatpak remotes --system`? I don't think the system service has to do anything regarding user remotes and vice-versa, but correct me if I'm wrong.

Reply:
Yeah, true. I modified the behavior of this a bit afterwards. You can see how it looks in bash module now.

Reply:
Also, here's how to do the awk checks with Nushell
```nu
~> echo "fedora" | split row "\n" | any {|remote| $remote == "fedora" or $remote == "fedora-testing"}
true
~> flatpak remotes --columns name | split row "\n" | any {|remote| $remote == "fedora" or $remote == "fedora-testing"}
false
~> flatpak remotes --columns name
Name
flathub
gnome-nightly
```

(the `or` is there just for illustration purposes, it can be removed if/when the actual logic requires)

Reply:
[6510afa](https://github.com/blue-build/modules/pull/336/commits/6510afa431e19dfa02de51f1277bfaf846143356) implements this partly. Please notify me of any changes required, or just do them yourself.

Reply:
I'm just now looking at https://github.com/blue-build/modules/pull/345/files

You could re-implement that in Nushell if you want to get your feet wet. The docs are great IMO.